tests/linearizability/linearizability_test.go

```diff
@@ -96,23 +97,31 @@ func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProce
 		t.Fatal(err)
 	}
 	defer clus.Close()
-	ctx, cancel := context.WithCancel(ctx)
+	trafficCtx, trafficCancel := context.WithCancel(ctx)
 	go func() {
-		defer cancel()
-		err := triggerFailpoints(ctx, t, clus, failpoint)
-		if err != nil {
-			t.Error(err)
-		}
+		defer trafficCancel()
+		triggerFailpoints(ctx, t, clus, failpoint)
+		// Wait second to collect traffic after triggering last failpoint.
+		time.Sleep(time.Second)
 	}()
-	operations := simulateTraffic(ctx, t, clus, traffic)
+	watchCtx, watchCancel := context.WithCancel(ctx)
+	var operations []porcupine.Operation
+	go func() {
+		defer watchCancel()
+		operations = simulateTraffic(trafficCtx, t, clus, traffic)
+		// Wait second to collect watch events after all traffic was sent.
+		time.Sleep(time.Second)
+	}()
+	events := watchClusterChanges(watchCtx, t, clus)
 	err = clus.Stop()
 	if err != nil {
 		t.Error(err)
 	}
+	validateEventsMatch(t, events)
```

Comment:
Since all the watch events are cached, so I guess you will almost double the memory usage? Do you have any data related to the memory usage diff to share?

Reply:
Don't understand the relevance. I didn't collect any data. What's the reason to collect memory usage here?

Reply:
See line 33 in watch.go, all watch events of all members are cached. It definitely causes more memory usage, and it's the reason why I raised this comment.  
```
memberEvents := make([][]watchEvent, len(clus.Procs))
```

Reply:
Are you asking why we keep events from all members? It's to compare if the history matches.

Reply:
No, I was concerning about the memory usage. Usually github runner has limited resource, I am not sure whether it will cause more flaky test.

If you do not see any issue after multiple runs, then I am OK.

Reply:
I'm not worried as we can leverage bigger nodes provided by CNCF if needed.